tests/Backend/Snowflake/ImportTypedTableTest.php

```diff
@@ -22,6 +24,108 @@ public function setUp(): void
         $this->initEmptyTestBucketsForParallelTests();
     }
 
+    /**
+     * test _timestamp changing in typed tables
+     */
+    public function testImportDeduplication(): void
+    {
+        $bucketId = $this->getTestBucketId();
+        $payload = [
+            'name' => 'dedup',
+            'primaryKeysNames' => ['id'],
+            'columns' => [
+                ['name' => 'id', 'basetype' => 'INTEGER'],
+                ['name' => 'name', 'basetype' => 'STRING'],
+            ],
+        ];
+        $tableId = $this->_client->createTableDefinition($bucketId, $payload);
+        $workspaces = new Workspaces($this->workspaceSapiClient);
+        $workspace = $this->initTestWorkspace(self::BACKEND_SNOWFLAKE);
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+
+        // import table with duplicates
+        $csvFile = new CsvFile(__DIR__ . '/../../_data/languages.duplicates.csv');
+        $this->_client->writeTableAsync(
+            $tableId,
+            $csvFile,
+            [
+                'incremental' => false,
+            ]
+        );
+        $workspaces->cloneIntoWorkspace($workspace['id'], [
+            'input' => [
+                [
+                    'source' => $tableId,
+                    'destination' => 'languages',
+                ],
+            ],
+        ]);
+        $firstLoadData = $backend->fetchAll('languages', \PDO::FETCH_ASSOC);
+        // there are 3 rows 0,24,26
+        $this->assertCount(3, $firstLoadData);
+        $firstLoadZeroRow = null;
+        $firstLoadOtherRows = [];
+        foreach ($firstLoadData as $row) {
+            if ($row['id'] === '0') {
+                $firstLoadZeroRow = $row;
+                continue;
+            }
+            $firstLoadOtherRows[] = $row;
+        }
+        $this->assertNotNull($firstLoadZeroRow);
+
+        // import same data second time incrementally
+        // file
+        // - is missing row id 0
+        // - has new value for id 26
+        // - has same value for id 24
+        // - has new ids 1,11,25
+        $csvFile = new CsvFile(__DIR__ . '/../../_data/languages.more-rows_not_all_old.csv');
```

Comment:
nechces ten soubor prejmenovat? na neco jako `languages.more-rows_without_zero.csv`. Tech more-rows je tam vic a at je to zrejme co je ucel toho souboru. Jinak to vypada jako nejakej backup

Reply:
cajk done